viewing_party/party.py

```diff
@@ -1,23 +1,162 @@
 # ------------- WAVE 1 --------------------
+from logging.config import dictConfig
+
 
 def create_movie(title, genre, rating):
-    pass
+    if title == None or genre == None or rating == None:
+        return None
+    else:
+        movie ={"title": title,
+        "genre": genre,
+        "rating": rating,
+
+        }
+    
+    return movie
+
+def add_to_watched(user_data, movie):
+    user_data = {}
+    user_data["watched"] = [movie]
+
+    return user_data
+
+def add_to_watchlist(user_data, movie):
+    user_data["watchlist"] = [movie]
+
+    return user_data
+
+def watch_movie(user_data, title):
+    watchlist = user_data["watchlist"]
+    for item in range(len(watchlist)):
+        if watchlist[item]["title"] == title:
+            user_data["watched"].append(watchlist[item])
+            del watchlist[item]
+
+    return user_data
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
+def get_watched_avg_rating(user_data):
+    sum = 0
+    num_of_ratings = 0
+    watched_list = user_data["watched"]
+    for dictionary in watched_list:
+        sum += dictionary["rating"]
+        num_of_ratings += 1
+    if num_of_ratings == 0:
+        return 0
+    else:
+        rating_avg = sum / num_of_ratings
 
+    return rating_avg
+
+def get_most_watched_genre(user_data):
+    if len(user_data["watched"]) < 1:
+        return None
+    genres = []
+    for dictionary in user_data["watched"]:
+        genres.append(dictionary["genre"])
+    genre_set = set(genres)
+    top_genre_dict = {genre : genres.count(genre) for genre in genre_set}
+    counter = 0
+    for value in top_genre_dict.values():
+        if counter < value:
+            counter = value
+            top_genre = list(top_genre_dict.keys())[list(top_genre_dict.values()).index(counter)]
+        
+    return top_genre
 
 # -----------------------------------------
 # ------------- WAVE 3 --------------------
 # -----------------------------------------
+def get_unique_watched(user_data):
+    user_movie_list = user_data["watched"]
+    friend_watched_list = user_data["friends"]
+    user_unique_movies = []
+    #------------------user data-------------------
+    for dictionary in user_movie_list:
+        if dictionary["title"] not in user_unique_movies:
+            user_unique_movies.append(dictionary)
+    
+    #--------------friend watched items-------------
+    for friends_dict in friend_watched_list:
+        for watched_list in friends_dict:
+            for dict in friends_dict[watched_list]:
+                for dictionary in user_unique_movies:
+                    if dictionary["title"] == dict["title"]:
+                        user_unique_movies.remove(dictionary)

```

Comment:
So I noticed you are using five for loops in this function. You could simplify this by iterating over the user watched list and checking if the movie exist within `user_unique_movies`. You can then add or remove to `user_unique_movies` based on if the user's friends have watched it. Something like this:
```
    for watched in user_data["watched"]:
        if watched in user_unique_movies:
            pass
        else:
            user_unique_movies.append(watched)

    for friend in user_data["friends"]:
        for movie in friend["watched"]:
            if movie in final_list:
                user_unique_movies.remove(movie)
            else:
                pass

    return user_unique_movies
```